### PR TITLE
fix(TDP-5898) Previous preparation results are no longer used

### PR DESCRIPTION
**Pull request by skermabon**
* Use the executeSampleStrategy any time, to cache when we send a HTTP 202 to StreamingResponseBody otherwise.
* Fix the accept of OptimizedExportStrategy
* Fix the ApplyPreparationToContentStrategy

**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-5898

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/TransformationService.java [hunk 1]
```diff
@@ -225,20 +226,20 @@ public StreamingResponseBody execute(@ApiParam(
...
-            if (!contentCache.has(cacheKey)) {
-                preparationExportStrategy.performPreparation(completeParameters, new NullOutputStream());
-            }
+        // Async behavior
+        final ConditionalTest conditionalTest = applicationContext.getBean(GetPrepContentAsyncCondition.class);
+        if (conditionalTest.apply(completeParameters)) {
+            // write to cache
+            executeSampleExportStrategy(completeParameters).writeTo(new NullOutputStream());
+            return outputStream -> {
+                outputStream.close();
```
Comment (fhuaulme) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/TransformationService.java]: I'm not sure closing the output stream (provided by app container) is a good idea.

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/export/OptimizedExportStrategy.java [hunk 2]
```diff
@@ -271,7 +271,7 @@ private OptimizedPreparationInput invoke() throws IOException {
                 LOGGER.debug("Not enough steps ({}) in preparation.", steps.size());
```
Comment (fhuaulme) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/export/OptimizedExportStrategy.java]: I think I'd rather prefer a code such as:
```java
final List<String> steps = new ArrayList<>(preparation.getSteps());
            if (steps.size() <= 2) {
                LOGGER.debug("Not enough steps ({}) in preparation.", steps.size());
                return null;
            }
            if (StringUtils.equals("head", stepId) || StringUtils.isEmpty(stepId)) {
                version = steps.get(steps.size() - 1);
                previousVersion = steps.get(steps.size() - 2);
            } else {
                version = stepId;
                previousVersion = steps.get(preparation.getSteps().indexOf(version) - 1);
            }
```

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/TransformationService.java (hunk 1 above)
Comment (skermabon) [reply dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/TransformationService.java]: ok, I remove it

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/export/OptimizedExportStrategy.java (hunk 2 above)
Comment (skermabon) [reply dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/export/OptimizedExportStrategy.java]: ok, so we can use optimized strategy with not only one step to add.

**Pull request merged**